src/limiter/limiter.go

```diff
@@ -165,7 +165,11 @@ func (l *Limiter) HitRateLimit(requestWeight int) error {
 			return err
 		}
 
-		if err = l.adjustConfig(requestWeight, c); err != nil {
+		if len(wait) == 0 {
+			wait = append(wait, 0)
```

Comment:
this seems "hacky". I don't understand the use case for `...int64` but it seems like since you're calling `wait[0]` below this really shouldn't be variadic. The logic seems to be like:

- If nothing is passed for wait, don't wait
- If something is passed for wait, do wait.

It seems like the amount to wait should be handled once when creating the config, rather than each time this is called as the caller would have to have access to the config to know how long to wait.